episodes/preparing_for_machine_learning.md

```diff
@@ -0,0 +1,438 @@
+rows = 1
+columns = 3
+
+# Adds a subplot at the 1st position
+fig.add_subplot(rows, columns, 1)
+# showing image
+plt.imshow(image_g)
+plt.axis('off')
+plt.title("Normal 1")
+  
+# add a subplot at the 2nd position
+fig.add_subplot(rows, columns, 2)
+# showing image
+plt.imshow(image_b)
+plt.axis('off')
+plt.title("Cardiomegaly")
+
+# add a subplot at the 3nd position
+fig.add_subplot(rows, columns, 3)
+# showing image
+plt.imshow(image_y)
+plt.axis('off')
+plt.title("Normal 2")
+```
+
+![](fig/cxr_display_mip.png){alt='CXR examples'}
+
+
+
+::::::::::::::::::::::::::::::::::::: challenge 
+
+## Thought Challenge: Can you see some problems in the following scenario?
+
+Imagine you got the above images and many more because you have been assigned to make an algorithm for cardiomegaly detection so patients can be notified if by chance their X-rays acquired by the hospital for any purpose shows cardiomegaly. (This would be an example of using machine learning for opportunistic screening.) You are given one dataset of healthy (no cardiomegaly) patients who went to an outpatient clinic in a very poor area staffed with first year radiography students. Those patients were being checked for tuberculosis (TB). You also got chest X-rays of cardiomegaly patients from an extremely prestigeous tertiary inpatient hospital. 
+
+:::::::::::::::::::::::: solution 
+
+## Solution
+ 
+All of the following may be potential problems:
+
+- We may have different quality images from different machines for our healthy versus diseased in this scenario, this could be called a "batch effect" creating bias
+- In the presigous hospital many X-rays may be supine, will these combine well with standing APs?
+- We could suspect labelling may not be entirely accurate in the clinic because has by lower level staff 
+
+:::::::::::::::::::::::::::::::::
+
+::::::::::::::::::::::::::::::::::::::::::::::::
+
+
+::::::::::::::::::::::::::::::::::::: challenge 
+
+## Code Challenge: Using skimage.transform.rotate
+
+Use `skimage.tranform.rotate` to produce two realistic augmented images (name them `new_pic1` and `new_pic2`) from the given 'normal' image we put in `image_g` variable, and then apply what you percieve as the two most critical algorithms to make them ready for classic supervised machine learning in one bit of code. 
+
+Hint: look at the shape of the cardiomegaly image in more ways than one. What will harsh lines do when we do ML?
+
+:::::::::::::::::::::::: solution 
+
+## Solution
+ 
+```python
+# figure out how much to cut on sides
+print("cut top/bottom:", (image_b.shape[0] - image_g.shape[0])/2)
+cut_top_bottom = abs(round((image_b.shape[0] - image_g.shape[0])/2))
+# figure our how much to cut on top and bottom
+print("cut sides:",(image_b.shape[1] - image_g.shape[1])/2)
+
+```
+
+```output
+cut top/bottom: -119.0
+cut sides: -208.5
+```
+
+
+```python
+cut_sides = abs(round((image_b.shape[1] - image_g.shape[1])/2))
+list_images = [image_g, new_pic1, new_pic2]
```

Comment:
The code to generate `new_pic1` and `new_pic2` needs be added to the solution.

Reply:
Actually I rewrote it not requesting new names.